src/app/wallets/get-lightning-fee.ts

```diff
@@ -0,0 +1,161 @@
+  decodedInvoice: LnInvoice
+  paymentAmount: Satoshis
+  logger: Logger
+}): Promise<Satoshis | ApplicationError> => {
+  const { destination, paymentHash, milliSatsAmount } = decodedInvoice
+
+  const walletId = await walletIdFromPublicId(walletPublicId)
+  if (walletId instanceof Error) return walletId
+
+  const balance = await getBalanceForWallet({ walletId, logger })
+  if (balance instanceof Error) return balance
+  if (balance < paymentAmount) {
+    return new InsufficientBalanceError(
+      `Payment amount '${paymentAmount}' exceeds balance '${balance}'`,
+    )
+  }
+
+  const lndService = LndService()
+  if (lndService instanceof Error) throw lndService
+  if (lndService.isLocal(destination)) {
+    return toSats(0)
+  }
+
+  const key = CachedRouteLookupKeyFactory().create({
+    paymentHash,
+    milliSats: milliSatsAmount,
+  })
+  const routeFromCache = await RoutesCache().findByKey(key)
+  const validCachedRoute = !(routeFromCache instanceof Error)
+  if (validCachedRoute) return toSats(routeFromCache.route.fee)
+
+  const maxFee = LnFeeCalculator().max(paymentAmount)
+
+  const rawRoute = await lndService.findRouteForInvoice({ decodedInvoice, maxFee })
+  if (rawRoute instanceof Error) return rawRoute
+
+  const routeToCache = { pubkey: lndService.defaultPubkey(), route: rawRoute }
+  const cachedRoute = await RoutesCache().store({ key, routeToCache })
+  if (cachedRoute instanceof Error) return cachedRoute
+
+  return toSats(rawRoute.fee)
+}
+
+const noAmountProbeForFee = async ({
+  walletPublicId,
+  decodedInvoice,
+  paymentAmount,
+  logger,
+}: {
+  walletPublicId: WalletPublicId
+  decodedInvoice: LnInvoice
+  paymentAmount: Satoshis
+  logger: Logger
+}): Promise<Satoshis | ApplicationError> => {
+  const { destination, paymentHash } = decodedInvoice
+
+  const walletId = await walletIdFromPublicId(walletPublicId)
+  if (walletId instanceof Error) return walletId
+
+  const balance = await getBalanceForWallet({ walletId, logger })
+  if (balance instanceof Error) return balance
+  if (balance < paymentAmount) {
+    return new InsufficientBalanceError(
+      `Payment amount '${paymentAmount}' exceeds balance '${balance}'`,
+    )
+  }
+
+  const lndService = LndService()
+  if (lndService instanceof Error) throw lndService
+  if (lndService.isLocal(destination)) {
+    return toSats(0)
+  }
+
+  const key = CachedRouteLookupKeyFactory().create({
+    paymentHash,
+    milliSats: toMilliSatsFromNumber(paymentAmount),
+  })
+  const routeFromCache = await RoutesCache().findByKey(key)
+  const validCachedRoute = !(routeFromCache instanceof Error)
+  if (validCachedRoute) return toSats(routeFromCache.route.fee)
+
+  const maxFee = LnFeeCalculator().max(paymentAmount)
+
+  const rawRoute = await lndService.findRouteForNoAmountInvoice({
+    decodedInvoice,
+    maxFee,
+    amount: paymentAmount,
+  })
+  if (rawRoute instanceof RouteNotFoundError) return rawRoute
+  if (rawRoute instanceof Error) return rawRoute
```

Comment:
is there any benefits to keeping those both lines, versus just using the latter?

Reply:
Oversight here from earlier version of the PR, fixed